independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/interceptors/bindings/repeatable/InheritedRepeatableInterceptorBindingTest.java

```diff
@@ -0,0 +1,148 @@
+package io.quarkus.arc.test.interceptors.bindings.repeatable;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.enterprise.context.ApplicationScoped;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InterceptorBinding;
+import javax.interceptor.InvocationContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+/**
+ * Tests usage of inherited repeating interceptor binding.
+ */
+public class InheritedRepeatableInterceptorBindingTest {
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(MyBinding.class, MyBinding.List.class,
+            SuperclassWithMethodLevelBindings.class, MethodInterceptedBean.class,
+            SuperclassWithClassLevelBindings.class, ClassInterceptedBean.class, IncrementingInterceptor.class);
+
+    @Test
+    public void methodLevelInterceptor() {
+        MethodInterceptedBean bean = Arc.container().instance(MethodInterceptedBean.class).get();
+
+        assertEquals(10, bean.foo());
+        assertEquals(21, bean.foobar());
+        assertEquals(30, bean.foobaz());
+        assertEquals(41, bean.foobarbaz());
+        assertEquals(50, bean.nonannotated());
+    }
+
+    @Test
+    public void classLevelInterceptor() {
+        ClassInterceptedBean bean = Arc.container().instance(ClassInterceptedBean.class).get();
+
+        assertEquals(10, bean.foo());
```

Comment:
Shouldn't the expected value be `11` because `ClassInterceptedBean#foo()` declares `@MyBinding("foo")` and `@MyBinding("bar")` is inherited from the superclass?

Reply:
Exactly the kind of feedback I was looking for!

My understanding is that if you declare an interceptor binding on a method, interceptor bindings of the same type coming from the class and all superclasses (and even stereotypes I think) are ignored.

This is per https://jakarta.ee/specifications/interceptors/2.0/interceptors-spec-2.0.html#binding_an_interceptor_to_a_component:

> An interceptor binding declared on a method or constructor replaces an interceptor binding of the same type declared at class level or inherited from a superclass [7].

Reply:
You're right but it's not very intuitive :-(. This interceptors rule is another one that was created before repeatable annotations were introduced...

Reply:
Yea I totally agree, the interceptors spec needs an update wrt repeatable annotations and default interface methods.

Reply:
Yet another reason to ditch the support of the class-level repeatable bindings...

Reply:
Agree. The test is "correct" per spec but it doesn't follow common sense.
Since the specification is blurred in this regard (e.g. repeatable ann), we may IMO choose to go either way.
FTR I did another quick test in Weld and Weld sides with Ladislav on this one as it also chooses not to inherit repeating annotation if declared on the current class.